src/main/resources/static/js/MainWindow.js

```diff
@@ -1,670 +1,769 @@
-        delimiter: currentDelimiter,
-        numberOfHeaderlines: parseInt(currentHeaderLines),
-        timezone: $("#selecttime option:selected").attr("data-value"),
-        dateTime: date,
-        streamData: streams,
-        mapOfMagicNumbers: map,
-        dataType: filetype,
-        frostURL: url
-    };
-
-    var jsoncfg = JSON.stringify(formData, null, 4);
-
-    $.ajax({
-            type: "POST",
-            url: "config/create",
-            contentType: "application/json",
-            data: jsoncfg,
-            dataType: "json",
-            error: function (e) {
-                addToLog(e.responseText);
-            },
-            success: function (e) {
-                var option = new Option(e.name, e.name, null, null);
-                option.setAttribute("data-value", JSON.stringify(e, null, 4));
-                $("#configs").append(option).val(e.name).trigger("change");
-                addToLog("Configuration saved.");
-            }
-        }
-    );
+	var currentInput;
+	var parsed;
+	var stop = false;
+	var url = document.querySelector("#frostserverurl").value;
+
+	try {
+		new RegExp(currentDelimiter);
+		if (currentDelimiter === null || currentDelimiter === "") {
+			currentDelimiter = "\n";
+		}
+	} catch (e) {
+		currentDelimiter = "\n";
+	}
+
+	if (!/^0*[0-9]{1,4}$/.test(currentHeaderLines)) {
+		currentHeaderLines = 0;
+	}
+
+	var date = [];
+	$("#timeTable")
+			.find("tbody tr")
+			.each(
+					function() {
+						var obj = {}, $td = $(this).find("td");
+						currentInput = $td.eq(1).find("input").val();
+						if ((currentInput === null || currentInput === "")
+								&& isExcel === false) {
+							$
+									.notify(
+											{
+												message : "Please specify the date format (no empty strings allowed)."
+											},
+											{
+												allow_dismiss : true,
+												type : "info",
+												placement : {
+													from : "top",
+													align : "left"
+												},
+												animate : {
+													enter : "animated fadeInDown",
+													exit : "animated fadeOutUp"
+												}
+											});
+							stop = true;
+							return false;
+						}
+						obj["string"] = currentInput;
+						currentInput = $td.eq(0).find("input").val();
+						if (currentInput === null || currentInput === "") {
+							$
+									.notify(
+											{
+												message : "Please specify the column where the date can be found (no empty field allowed) before saving the configuration."
+											},
+											{
+												allow_dismiss : true,
+												type : "info",
+												placement : {
+													from : "top",
+													align : "left"
+												},
+												animate : {
+													enter : "animated fadeInDown",
+													exit : "animated fadeOutUp"
+												}
+											});
+							stop = true;
+							return false;
+						}
+						parsed = parseInt(currentInput, 10);
+
+						if (!(currentInput == parsed) || parsed < 0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FraunhoferIOSB/ChillImport/pullRequest?prid=4781409)